src/core/ipe-setup.c

```diff
@@ -0,0 +1,165 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "copy.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "ipe-setup.h"
+#include "path-util.h"
+
+#define IPE_SECFS_DIR "/sys/kernel/security/ipe"
+#define IPE_SECFS_NEW_POLICY IPE_SECFS_DIR "/new_policy"
+#define IPE_SECFS_POLICIES IPE_SECFS_DIR "/policies/"
+#define IPE_POLICY_PATH "/etc/ipe/ipe-policy.p7b"
+
+int ipe_setup(void) {
+#if ENABLE_IPE
+        _cleanup_free_ char *policy_name = NULL, *policy_version = NULL, *policy_existing_version = NULL,
+                *policy_existing_version_file = NULL, *policy_activate = NULL, *output_path_free = NULL;
+        _cleanup_fclose_ FILE *finput = NULL;
+        _cleanup_close_ int input = -EBADF, output = -EBADF;
+        const char *output_path;
+        int r;
+
+        if (access(IPE_SECFS_DIR, F_OK) < 0) {
+                log_debug_errno(errno, "IPE support is disabled in the kernel, ignoring: %m");
+                return 0;
+        }
+
+        if (access(IPE_POLICY_PATH, F_OK) < 0) {
```

Comment:
This seems unnecessary? Instead, just check -ENOENT for open() call below?

Reply:
This is in the most critical and hottest path, right at boot where everything is sequential so this delays everything else, so it's worth doing the lightest possible smoke test first and get out of the way as quickly as possible, as in the vast majority of the cases there won't be anything. IMA and SELinux setups do exactly the same thing.